Tests/MaintenanceTests/MaintenanceTests.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  MaintenanceTests.swift
+//
+//
+//  Created by Emad Ghorbania on 18/11/2022.
+//
+
+import XCTest
+@testable import Maintenance
+import Firebase
+final class MaintenanceTests: XCTestCase {
+
+    func testFullscreen() async throws {
+        let input = Maintenance.isThereAnyMaintenance(
+            fullscreen: true,
+            banner: false,
+            fullscreenDetail: Maintenance.Detail(title: "FullScreen", body: "Body Test"),
+            bannerDetail: Maintenance.Detail(title: "Banner", body: "Body Test"))
+        let expectedOutput = Maintenance.StatusType.fullscreen
+        XCTAssertEqual(input?.0, expectedOutput, "The output is not expected!")
+    }
+    func testBanner() async throws {
+        let input = Maintenance.isThereAnyMaintenance(
+            fullscreen: false,
+            banner: true,
+            fullscreenDetail: Maintenance.Detail(title: "FullScreen", body: "Body Test"),
+            bannerDetail: Maintenance.Detail(title: "Banner", body: "Body Test"))
+        let expectedOutput = Maintenance.StatusType.banner
+        XCTAssertEqual(input?.0, expectedOutput, "The output is not expected!")
+    }
+    
+    func testFullscreenDetails() async throws {
+        let input = Maintenance.isThereAnyMaintenance(
+            fullscreen: true,
+            banner: false,
+            fullscreenDetail: Maintenance.Detail(title: "FullScreen", body: "Body Test"),
+            bannerDetail: Maintenance.Detail(title: "Banner", body: "Body Test"))
+        let expectedOutput = Maintenance.Detail(title: "FullScreen", body: "Body Test")
+        XCTAssertEqual(input?.1, expectedOutput, "The output is not expected!")
+    }
+    func testBannerDetails() async throws {
+        let input = Maintenance.isThereAnyMaintenance(
+            fullscreen: false,
+            banner: true,
+            fullscreenDetail: Maintenance.Detail(title: "FullScreen", body: "Body Test"),
+            bannerDetail: Maintenance.Detail(title: "Banner", body: "Body Test"))
+        let expectedOutput = Maintenance.Detail(title: "Banner", body: "Body Test")
+        XCTAssertEqual(input?.1, expectedOutput, "The output is not expected!")
+    }
```

Comment:
An idea: Couldn't you merge these test with `testFullscreen` and `testBanner`? By merging them you would have less maintenance, because the input data for the tests are the same.

Reply:
How would you think that they can be merged?

Reply:
```suggestion
    func testFullscreen() async throws {
        let input = Maintenance.isThereAnyMaintenance(
            fullscreen: true,
            banner: false,
            fullscreenDetail: Maintenance.Detail(title: "FullScreen", body: "Body Test"),
            bannerDetail: Maintenance.Detail(title: "Banner", body: "Body Test"))
        
        XCTAssertEqual(input?.0, Maintenance.StatusType.fullscreen, "The output is not expected!")
        XCTAssertEqual(input?.1, Maintenance.Detail(title: "FullScreen", body: "Body Test"), "The output is not expected!")
    }
    func testBanner() async throws {
        let input = Maintenance.isThereAnyMaintenance(
            fullscreen: false,
            banner: true,
            fullscreenDetail: Maintenance.Detail(title: "FullScreen", body: "Body Test"),
            bannerDetail: Maintenance.Detail(title: "Banner", body: "Body Test"))
            
        XCTAssertEqual(input?.0, Maintenance.StatusType.banner, "The output is not expected!")
        XCTAssertEqual(input?.1, Maintenance.Detail(title: "Banner", body: "Body Test"), "The output is not expected!")
    }
```